node/src/service.rs

```diff
@@ -170,7 +173,7 @@ where
 	let transaction_pool = params.transaction_pool.clone();
 	let mut task_manager = params.task_manager;
 	let import_queue = params.import_queue;
-	let (block_import, pending_transactions, filter_pool) = params.other;
+	let (_block_import, pending_transactions, filter_pool) = params.other;
```

Comment:
Why do we return the block import from `new_partial` at all?

Reply:
Okay, so we do need it. I commented where below. But also, check out how we do the pattern matching below around line ~395. I think it is more elegant :)

Reply:
Also, this is a perfect example of why we need to run integration tests against the real parachain node. Even though we've reduced duplicated code to just these two functions, this is a bug that would not have manifested in the dev service.

Reply:
Some background reading about the block import pipeline https://substrate.dev/docs/en/knowledgebase/advanced/block-import#docsNav for the curious.